### PR TITLE
Don't duplicate input names in outputs

### PR DESCRIPTION
**Pull request by cole-h**

### CODE REVIEW

**Comment by cole-h** [comment]
Pushed the test first so you can see the failure for yourself.

File: src/cli/cmd/add/flake.rs [hunk 1]
```diff
@@ -275,6 +275,15 @@ impl AttrType {
         flake_contents: &str,
     ) -> color_eyre::Result<String> {
         let mut new_flake_contents = flake_contents.to_string();
+
+        if head
+            .arguments
+            .iter()
+            .any(|arg| &*arg.identifier == flake_input_name)
+        {
```
Comment (Hoverbear) [inline src/cli/cmd/add/flake.rs:283]: I'd normally add a `tracing::trace!("Found existing duplicate input")` here so when I got very confused later why my code was exiting early I might actually notice.

**Review by Hoverbear — APPROVED** [review]

File: src/cli/cmd/add/flake.rs (hunk 1 above)
Comment (cole-h) [reply src/cli/cmd/add/flake.rs:283]: We'd have to setup tracing first...! I'll make a ticket for that.

**Review by grahamc — COMMENTED** [review]
Nice, fixes it for me.

**Pull request merged**